src/test/java/seedu/address/model/moduleclass/ModuleNameContainsKeywordsPredicateTest.java

```diff
@@ -0,0 +1,77 @@
+package seedu.address.model.moduleclass;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_NAME_CS2103T_TUTORIAL;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.testutil.ModuleClassBuilder;
+
+public class ModuleNameContainsKeywordsPredicateTest {
+
+    @Test
+    public void equals() {
+        List<String> firstPredicateKeywordList = Collections.singletonList("first");
+        List<String> secondPredicateKeywordList = Arrays.asList("first", "second");
+
+        ModuleNameContainsKeywordsPredicate firstPredicate =
+                new ModuleNameContainsKeywordsPredicate(firstPredicateKeywordList);
+        ModuleNameContainsKeywordsPredicate secondPredicate =
+                new ModuleNameContainsKeywordsPredicate(secondPredicateKeywordList);
+
+        // same object -> returns true
+        assertTrue(firstPredicate.equals(firstPredicate));
+
+        // same values -> returns true
+        ModuleNameContainsKeywordsPredicate firstPredicateCopy =
+                new ModuleNameContainsKeywordsPredicate(firstPredicateKeywordList);
+        assertTrue(firstPredicate.equals(firstPredicateCopy));
+
+        // different types -> returns false
+        assertFalse(firstPredicate.equals(1));
+
+        // null -> returns false
+        assertFalse(firstPredicate.equals(null));
+
+        // different module class -> returns false
+        assertFalse(firstPredicate.equals(secondPredicate));
+    }
+
+    @Test
+    public void test_nameContainsKeywords_returnsTrue() {
+        // One keyword
```

Comment:
start with lower case, e.g

```suggestion
        // one keyword
```